File: src/main/java/com/codethen/telegram/lanxatbot/exception/InlineQueryException.java
package com.codethen.telegram.lanxatbot.exception;

import org.telegram.telegrambots.meta.api.objects.inlinequery.InlineQuery;

public class InlineQueryException extends RuntimeException {

    private final InlineQuery inlineQuery;

    public InlineQueryException(InlineQuery inlineQuery, String message) {
        super(message);
        this.inlineQuery = inlineQuery;
    }

    public InlineQueryException(InlineQuery inlineQuery, String message, Throwable cause) {
        super(message, cause);
        this.inlineQuery = inlineQuery;
    }

    public InlineQuery getQuery() {
        return inlineQuery;
    }
}

File: src/main/kotlin/com/codethen/telegram/lanxatbot/exception/InlineQueryException.kt
package com.codethen.telegram.lanxatbot.exception

import org.telegram.telegrambots.meta.api.objects.inlinequery.InlineQuery

class InlineQueryException : RuntimeException {

    val query: InlineQuery

    constructor(query: InlineQuery, message: String): super(message) {
        this.query = query
    }

    constructor(query: InlineQuery, message: String, cause: Throwable): super(message, cause) {
        this.query = query
    }
}